src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/RegisterRenewConfirm.cs

```diff
@@ -73,5 +73,64 @@ public async Task Test()
                     change.CourseName,
                 });
         }
+
+
+        [Test]
+        public async Task ConfirmingCommitmentStatementConcurrentToApprovedChangeConfirmsCorrectStatement()
+        {
+            var factory = Bindings.Api.CreateApiFactory();
+            var client = factory.CreateClient();
+            var db = Bindings.Database.CreateDbContext();
+
+            var create = fixture.Build<CreateRegistrationCommand>()
+                .With(p => p.Email, (MailAddress adr) => adr.ToString())
+                .Create();
+
+            var r1 = await client.PostValueAsync("apprenticeships", create);
+            r1.EnsureSuccessStatusCode();
+
+            var verify = fixture.Build<VerifyRegistrationCommand>()
+                .With(x => x.ApprenticeId, create.ApprenticeId)
+                .With(p => p.Email, create.Email)
+                .Create();
+
+            var r2 = await client.PostValueAsync("registrations", verify);
+            r2.EnsureSuccessStatusCode();
+
+            var (r3, apprenticeships) = await client.GetValueAsync<List<ApprenticeshipDto>>($"apprentices/{create.ApprenticeId}/apprenticeships");
+            var apprenticeshipId = apprenticeships[0].Id;
+            var csId = apprenticeships[0].CommitmentStatementId;
+
+            var change = fixture.Build<ChangeApprenticeshipCommand>()
+                .With(x => x.ApprenticeshipId, create.ApprenticeshipId)
+                .With(p => p.Email, (MailAddress adr) => adr.ToString())
+                .With(p => p.ApprovedOn, (int days) => create.ApprovedOn.AddDays(days))
+                .Create();
+
+            var r5 = await client.PostValueAsync("apprenticeships/change", change);
+            r5.EnsureSuccessStatusCode();
+
+            var r4 = await client.PostValueAsync(
+                $"apprentices/{create.ApprenticeId}/apprenticeships/{apprenticeshipId}/{csId}/EmployerConfirmation",
+                new ConfirmEmployerRequest { EmployerCorrect = true });
+            r4.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var (r6, apprenticeships2) = await client.GetValueAsync<List<ApprenticeshipDto>>($"apprentices/{create.ApprenticeId}/apprenticeships");
+            r5.EnsureSuccessStatusCode();
+
+            apprenticeships2.Should().HaveCount(2)
+                .And.ContainEquivalentOf(new
+                {
+                    Id = apprenticeshipId,
+                    CommitmentStatementId = csId,
+                    EmployerCorrect = true,
+                })
+                .And.ContainEquivalentOf(new
+                {
+                    Id = apprenticeshipId,
+                    CommitmentStatementId = csId+1,
+                    EmployerCorrect = (bool?)null,
+                });
+        }
```

Comment:
I worry about this approach, we literally have to call several endpoints to get the data into a position that we can test this scenario. Can we not simply configure the models as we do in the other tests save them to the database and then run the step we are interested in.

As the system grows in complexity, the steps we are calling will start having functionality added and this will impact on this

Reply:
I see it as a positive to have at least some tests that are testing the project behaves correctly when used solely via the APIs.  If nothing else, it proves that the state left after call 1 is indeed a valid pre-state for call 2.  This would not be captured by tests that setup the state.

I have called these `workflow` tests (and moved them into a folder of that name) to highlight that they are testing the joined-up behaviour of the API.